ample/util/options_processor.py

```diff
@@ -90,6 +90,87 @@ def process_benchmark_options(optd):
+        try:
+            optd['gesamt_exe'] = ample_util.find_exe(optd['gesamt_exe'])
+        except ample_util.FileNotFoundError as e:
+            raise RuntimeError("Cannot find Gesamt executable: {0}".format(optd['gesamt_exe']))
+    # Ensemble options
+    if optd['cluster_method'] in [SPICKER_RMSD, SPICKER_TM]:
+        if not optd['spicker_exe']:
+            if optd['cluster_method'] == SPICKER_TM and optd['nproc'] > 1:
+                # We need to use the multicore version of SPICKER
+                optd['spicker_exe'] = 'spicker_omp' + ample_util.EXE_EXT
+            else:
+                optd['spicker_exe'] = 'spicker' + ample_util.EXE_EXT
+        try:
+            optd['spicker_exe'] = ample_util.find_exe(optd['spicker_exe'])
+        except ample_util.FileNotFoundError:
+            msg = "Cannot find spicker executable: {0}".format(optd['spicker_exe'])
+            exit_util.exit_error(msg)
+    elif optd['cluster_method'] in ['fast_protein_cluster']:
+        if not optd['fast_protein_cluster_exe']:
+            optd['fast_protein_cluster_exe'] = 'fast_protein_cluster'
+        try:
+            optd['fast_protein_cluster_exe'] = ample_util.find_exe(optd['fast_protein_cluster_exe'])
+        except ample_util.FileNotFoundError:
+            msg = "Cannot find fast_protein_cluster executable: {0}".format(optd['fast_protein_cluster_exe'])
+            exit_util.exit_error(msg)
+    elif optd['cluster_method'] in ['import', 'random', 'skip']:
+        pass
+    else:
+        raise RuntimeError("Unrecognised cluster_method: {0}".format(optd['cluster_method']))
+    if not optd['theseus_exe']:
+        optd['theseus_exe'] = os.path.join(os.environ['CCP4'], 'bin', 'theseus' + ample_util.EXE_EXT)
+    try:
+        optd['theseus_exe'] = ample_util.find_exe(optd['theseus_exe'])
+    except ample_util.FileNotFoundError:
+        raise RuntimeError("Cannot find theseus executable: {0}".format(optd['theseus_exe']))
+    # SCRWL - we always check for SCRWL as if we are processing QUARK models we want to add sidechains to them
+    if not optd['scwrl_exe']:
+        optd['scwrl_exe'] = os.path.join(os.environ['CCP4'], 'bin', 'Scwrl4' + ample_util.EXE_EXT)
+    try:
+        optd['scwrl_exe'] = ample_util.find_exe(optd['scwrl_exe'])
+    except ample_util.FileNotFoundError as e:
+        logger.info("Cannot find Scwrl executable: %s", optd['scwrl_exe'])
+        if optd['use_scwrl']:
+            raise (e)
+    if "subcluster_radius_thresholds" in optd and not optd["subcluster_radius_thresholds"]:
+        optd["subcluster_radius_thresholds"] = SUBCLUSTER_RADIUS_THRESHOLDS
+    # REM: This should really be disentangled and moved up to definition of all homologs options
+    # REM: but could cause confusion with defaults down here.
+    if "side_chain_treatments" in optd and not optd["side_chain_treatments"]:
+        if optd["homologs"]:
+            optd["side_chain_treatments"] = [POLYALA, RELIABLE, ALLATOM]
```

Comment:
`Enum`?

Reply:
Just a suggestion more than anything, might be easy to implement and much nicer to work with ...

Reply:
Yes but not right now ;)